libs/feature/record/src/lib/data-view-map/data-view-map.component.ts

```diff
@@ -151,24 +153,28 @@ export class DataViewMapComponent {
       )
     } else if (this.linkHelper.hasProtocolDownload(link)) {
       return fromPromise(
-        readDataset(link.url, 'geojson').then((features) => ({
-          type: MapContextLayerTypeEnum.GEOJSON,
-          data: {
-            type: 'FeatureCollection',
-            features,
-          },
-        }))
+        readDataset(this.proxy.getProxiedUrl(link.url), 'geojson').then(
+          (features) => ({
+            type: MapContextLayerTypeEnum.GEOJSON,
+            data: {
+              type: 'FeatureCollection',
+              features,
+            },
+          })
+        )
       )
     } else if (this.linkHelper.isEsriRestFeatureServer(link)) {
       const url = getEsriRestDataUrl(link, 'geojson')
       return fromPromise(
-        readDataset(url, 'geojson').then((features) => ({
-          type: MapContextLayerTypeEnum.GEOJSON,
-          data: {
-            type: 'FeatureCollection',
-            features,
-          },
-        }))
+        readDataset(this.proxy.getProxiedUrl(url), 'geojson').then(
```

Comment:
duplicated code, should the if just about the `url` var set ?

Reply:
yeah I agree it's bad... sorry, don't have much time now, we'll revisit this later on I guess